File: nodejs-nestjs/src/adapter/in/order.controller.ts
import { Controller, Post, Body, HttpCode, Logger } from '@nestjs/common';
import { CreateOrderUseCase } from '../../application/create-order.use-case';
import { CreateOrderInput } from '../../application/dto';

@Controller('api/orders')
export class OrderController {
  private readonly logger = new Logger(OrderController.name);

  constructor(private readonly useCase: CreateOrderUseCase) {
    this.logger.log = () => {};
  }

  @Post()
  @HttpCode(200)
  async create(@Body() input: CreateOrderInput) {
    return this.useCase.execute(input);
  }
}

File: java-springboot/src/main/java/com/benchmark/orders/adapter/in/OrderController.java
package com.benchmark.orders.adapter.in;

import com.benchmark.orders.application.CreateOrderUseCase;
import com.benchmark.orders.application.dto.CreateOrderInput;
import com.benchmark.orders.application.dto.CreateOrderOutput;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/orders")
public class OrderController {

    private final CreateOrderUseCase useCase;

    public OrderController(CreateOrderUseCase useCase) {
        this.useCase = useCase;
    }

    @PostMapping
    public ResponseEntity<?> createOrder(@RequestBody CreateOrderInput input) {
        try {
            CreateOrderOutput output = useCase.execute(input);
            return ResponseEntity.ok(output);
        } catch (IllegalArgumentException e) {
            return ResponseEntity.badRequest()
                    .body(Map.of("error", "validation_error", "message", e.getMessage()));
        }
    }
}
